src/files.rs

```diff
@@ -19,16 +19,32 @@ pub fn read_file(path: &Path) -> Fallible<Vec<u8>> {
     std::fs::read(&path).map_err(|err| format!("{}: {}", err, path.display()).into())
 }
 
+fn files_differ(old: &Path, new: &Path) -> Fallible<bool> {
+    if !(old.is_file() && new.is_file()) {
+        return Ok(true);
+    }
+
+    Ok(read_file(old)? != read_file(new)?)
+}
+
+pub fn file_differs(file: &Path, content: &str) -> Fallible<bool> {
+    if !file.is_file() {
```

Comment:
This looks redundant as reading the file would fail otherwise. Why not just turn failure-to-read into a change?